pybrowscap/loader/csv/__init__.py

```diff
@@ -43,45 +43,59 @@ def replace_defaults(line, defaults):
 
         """
         new_line = {}
-        for feature, value in line.iteritems():
+        try:
+            items = line.iteritems()
+        except AttributeError:
+            items = line.items()
+            
+        for feature, value in items:
             if value == 'default' or value == '':
-                value = defaults.get(feature, value)
+                value = defaults[feature]
             if value == 'true':
                 value = True
             if value == 'false':
                 value = False
             if feature == 'MinorVer' and value == '0':
-                value = defaults.get(feature, value)
+                value = defaults[feature]
             if feature == 'MajorVer' or feature == 'MinorVer':
                 try:
                     value = int(value)
                 except (ValueError, OverflowError):
                     value = 0
             if (feature == 'Version' or feature == 'RenderingEngine_Version') and value == '0':
-                value = defaults.get(feature, value)
-            if (feature == 'CSSVersion' or feature == 'AolVersion' or feature == 'Version' or
+                value = defaults[feature]
+            if (feature == 'CssVersion' or feature == 'AolVersion' or feature == 'Version' or
                 feature == 'RenderingEngine_Version' or feature == 'Platform_Version'):
                 try:
                     value = float(value)
                 except (ValueError, OverflowError):
                     value = float(0)
             new_line[feature.lower()] = value
         return new_line
+
     try:
-        with open(browscap_file_path, 'rb') as csvfile:
+        with open(browscap_file_path, 'r') as csvfile:
             log.info('Reading browscap source file %s', browscap_file_path)
             dialect = csv.Sniffer().sniff(csvfile.read(4096))
             csvfile.seek(0)
             log.info('Getting file version and release date')
             csvfile.readline()
-            line = csv.reader(StringIO(csvfile.readline())).next()
+            try:
+                # line = csv.reader(StringIO(csvfile.readline())).next()
```

Comment:
Can you pls provide some explanation for this change ?

Reply:
py3 not support next() method, but py2 and py3 both support this:

`import csv
with open('some.csv', newline='') as f:
    reader = csv.reader(f)
    for row in reader:
        print(row)
`